File: BOJ/seulah/7_week/좋다.py
import sys
input = sys.stdin.readline

n = int(input())
arr = sorted(list(map(int, input().split())))
good = 0

for i in range(n):
    left , right = 0, n-1
    target = arr[i]

    while left < right:
        if i == left:
            left+=1
            continue
        if i == right:
            right-=1
            continue

        total = arr[left]+arr[right]
        if target < total:
            right -=1
        elif target > total:
            left +=1
        else:
            good+=1
            break
print(good)

File: BOJ/jaehong/7_week/좋다.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");

const n = parseInt(input[0]);
const arr = input[1].split(" ").map(Number);

arr.sort((a, b) => a - b);

let count = 0;

for (let i = 0; i < n; i++) {
  let target = arr[i];
  let left = 0;
  let right = n - 1;

  while (left < right) {
    if (left === i) {
      left++;
      continue;
    }
    if (right === i) {
      right--;
      continue;
    }

    let sum = arr[left] + arr[right];

    if (sum === target) {
      count++;
      break;
    }

    if (sum < target) {
      left++;
    } else {
      right--;
    }
  }
}

console.log(count);
